File: ch05/CollectionView1/CollectionView1/EventCollectionViewCell.m
//
//  EventCollectionViewCell.m
//  CollectionView1
//
//  Created by MyZiyue on 2018/12/26.
//  Copyright © 2018 MyZiyue.com. All rights reserved.
//

#import "EventCollectionViewCell.h"

@implementation EventCollectionViewCell

-(id)initWithFrame:(CGRect)frame {
    self = [super initWithFrame:frame];
    
    if(self) {
        CGFloat cellWidth = self.frame.size.width;
        
        CGFloat imageViewWidth = 101;
        CGFloat imageViewHeight = 101;
        CGFloat imageViewTopView = 15;
        self.imageView = [[UIImageView alloc] initWithFrame:CGRectMake((cellWidth - imageViewWidth)/2, imageViewTopView, imageViewWidth, imageViewHeight)];
        [self addSubview:self.imageView];
        
        CGFloat labelWidth = 101;
        CGFloat labelHeight = 16;
        CGFloat labelTopView = 120;
        self.label = [[UILabel alloc] initWithFrame:CGRectMake((cellWidth - labelWidth)/2, labelTopView, labelWidth, labelHeight)];
        self.label.textAlignment = NSTextAlignmentCenter;
        self.label.font = [UIFont systemFontOfSize:13];
        [self addSubview:self.label];
    }
    return self;
}
@end

File: ch05/SCollectionView1/SCollectionView1/EventCollectionViewCell.swift
//
//  EventCollectionViewCell.swift
//  SCollectionView1
//
//  Created by MyZiyue on 2018/12/26.
//  Copyright © 2018 MyZiyue.com. All rights reserved.
//

import UIKit

class EventCollectionViewCell: UICollectionViewCell {
    var imageView:UIImageView!
    var label:UILabel!
    
    override init(frame:CGRect) {
        super.init(frame: frame)
        
        let cellWidth:CGFloat = self.frame.width
        let imageViewWidth:CGFloat = 101
        let imageViewHeight:CGFloat = 101
        let imageViewTopView:CGFloat = 15
        
        self.imageView = UIImageView(frame: CGRect(x:(cellWidth - imageViewWidth)/2, y:imageViewTopView, width: imageViewWidth, height:imageViewHeight))
        self.addSubview(self.imageView)
        
        let labelWidth:CGFloat = 101
        let labelHeight:CGFloat = 16
        let labelTopView:CGFloat = 120
        
        self.label = UILabel(frame: CGRect(x:(cellWidth - labelWidth)/2, y: labelTopView, width:labelWidth, height: labelHeight))
        self.label.textAlignment = .center
        self.label.font = UIFont.systemFont(ofSize: 13)
        self.addSubview(self.label)
    }
    
    required init?(coder aDecoder: NSCoder) {
        fatalError("init(coder:) has not been implemented")
    }
}
